youtube2zim/scraper.py

```diff
@@ -550,24 +572,71 @@ def get_slot():
 
         return overall_succeeded, overall_failed
 
-    @staticmethod
-    def download_video_files_batch(options, videos_ids):
+    def download_s3_cache(self, s3_key, local_video_path):
+        """returns True if successfully downloaded cache, otherwise False"""
+        try:
+            if self.use_any_optimized_version:
+                if not self.s3_storage.has_object(s3_key, self.bucket_name):
+                    return False
+            else:
+                if not self.s3_storage.has_object_matching_meta(
+                    s3_key,
+                    tag="encoding_version",
+                    value=ENCODER_VERSION,
+                    bucket_name=self.bucket_name,
+                ):
+                    return False
+            os.makedirs(os.path.dirname(local_video_path), exist_ok=True)
+            self.s3_storage.download_file(s3_key, local_video_path)
+        except Exception as ex:
+            logger.error(f"{self.bucket_name}/{s3_key} download failed with {str(ex)}")
+            return False
+        logger.info(
+            f"cache ({self.bucket_name}/{s3_key}) downloaded {local_video_path}"
+        )
+        return True
 
+    def upload_s3_cache(self, s3_key, local_video_path):
```

Comment:
rename to `upload_to_cache` and use `""" whether it successfully uploaded to cache """`